### PR TITLE
add X509 SAN extension vectors

### PR DESCRIPTION
**Pull request by reaperhulk**
refs #1743

### CODE REVIEW

**Comment by alex** [comment]
retest this please

File: vectors/cryptography_vectors/x509/custom/san_email_dns_ip_dirname_uri.pem [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ckBjcnlwdG9ncmFwaHkuaW+CD2NyeXB0b2dyYXBoeS5pb4cEfwAAAYcQAP8AAAAA
+AAAAAAAAAAAAAKQ0MDIxDjAMBgNVBAMMBWRpckNOMSAwHgYDVQQKDBdDcnlwdG9n
+cmFwaGljIEF1dGhvcml0eYYXaHR0cHM6Ly9jcnlwdG9ncmFwaHkuaW8wDQYJKoZI
+hvcNAQEFBQADggEBABpsK+icD/vLFrdzbg/qJghZ+DqAx4u8e3LmBQLKMgegrWhC
+wcXgqeG1m9BmasZv3UjS9v0oiwOQCmewopm7Q2F/IOzEqYmIoo56Y9//dou8+yvB
+10EA9Lu3k1aRQa2mAG83dGovPTEtbvdsMLhVP3udA5rADEPP4KC736/NrpPLj4pE
+thxhh0QJbGLQ9HouS6ic9WjYDhqxn6XJrOWSJSeRsVXn4AKAiYsQGNnSJGRZtjqN
+q24kUsuB7cAQSjWPKETl/UxTXTwhlStLg/s1aePRsOXromlCPEgRh+Noe/V7jFEG
+Hcbr0ocG/LezIT0EPOx1gErTQo6R9Rp+2DJEBFQ=
+-----END CERTIFICATE-----
```
Comment (chelseawinfree) [inline vectors/cryptography_vectors/x509/custom/san_email_dns_ip_dirname_uri.pem]: email:user@cryptography.io, DNS:cryptography.io, IP Address:127.0.0.1, IP Address:FF:0:0:0:0:0:0:0, DirName:/CN=dirCN/O=Cryptographic Authority, URI:https://cryptography.io   RSA Public Key: (2048 bit)  self signed by O=PyCA, CN=cryptography. (Verified)

File: vectors/cryptography_vectors/x509/custom/san_other_name.pem [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+srDiP+mqhCtTvubOJjOSZPP4dxganQbSZ2i0wJYBwo1YYFU3x9b2UHmmuVNi0kqj
+3fYkFEr6JmLJ/FalYPEc2ffs4ukSWCxwHGnbGvLt1LeteaufIWHiekuAEBkogtu3
+w4EKQUHGwkRv8gWyPG/jINPXAgMBAAGjIzAhMB8GA1UdEQQYMBagFAYDKgMEoA0W
+C0hlbGxvIFdvcmxkMA0GCSqGSIb3DQEBBQUAA4IBAQBFZHNeysAOnKD5Wq0vtZiw
+tfSJnbuxk4pqjvudSz2pH9abMgrfddlR0Zhb4VJx7+4DMGCNLhZHjNQhks670QZY
+MRq/x8Yk38pCVrF6pce87GHQSMdQ09GyknKakw0roN0iMzDy50RGiv+MXnjbiyjB
+lgqbs4rMoUI1CRTkid8yHuh+QUMETuQ17gKBCz+JdS12743ziZSNdjf9WKNzCEH8
+6YAFJUmu0xyVG1Rf1dgek/miLtB6sqUOAiC4IlTFCXu+c6UHd+d0OIZnaLNWiMDr
++5nh4BPuL/buvsLb0SOoKbnHZZYyBKFcdIf6iejsHpZ0lYmeBRqf0HlLUva1zWoM
+-----END CERTIFICATE-----
```
Comment (chelseawinfree) [inline vectors/cryptography_vectors/x509/custom/san_other_name.pem]: othername:<unsupported> self signed. (verified)

File: vectors/cryptography_vectors/x509/custom/san_registered_id.pem [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+7YxnFz1Yiik3/RqxbTL8ccc2F5VXXcwzc5A6EwtqJNVCG3NY/Ft5vYONVl20dyhV
+/7BzKA/Lv45FnSDyiGss/amNGodznEk/95QdeLkCAwEAAaOBrjCBqzAJBgNVHRME
+AjAAMB0GA1UdDgQWBBTnx8+v+mXQU9rQzOr3cgKfLE8UbDAfBgNVHSMEGDAWgBQK
+I5A8UXeub7z+dqK+7W7RQp2ntjALBgNVHQ8EBAMCBaAwEwYDVR0lBAwwCgYIKwYB
+BQUHAwEwLAYDVR0fBCUwIzAhoB+gHYYbaHR0cDovL3BhdGgudG8uY3JsL215Y2Eu
+Y3JsMA4GA1UdEQQHMAWIAyoDBDANBgkqhkiG9w0BAQUFAAOBgQBiOICo0oe0ndcE
++/QAV+rIMoDP5WeMw6/9o2r7sGTGJPmlMScef1+rSkdufv3SqehjWo8HYTF83BgD
+rwBtc1eU76bb3W7hhPv8pTAiQNsyYMDv2MKLMkL7UvKF+55acMGt/nJRN9zxdnIG
+ZuIUw9ZOtEByVZgu6BY0xseGOTEhvA==
+-----END CERTIFICATE-----
```
Comment (chelseawinfree) [inline vectors/cryptography_vectors/x509/custom/san_registered_id.pem]: X509v3 Subject Alternative Name:
                Registered ID:1.2.3.4

RSA 1024 issued from Rando Root. (verified)

**Comment by chelseawinfree** [comment]
These vectors have been verified using the openssl tool. All changes seem fine :+1:

**Pull request merged**